### DISCUSSION TITLE
next.js + browserfs + sql.js feature

### DISCUSSION THREAD

**Discussion by ericbets** [vercel/next.js#12374] [Help]
This is a feature proposal that I thought might fit in with the next.js design spirit. 

What if next.js had an internal filesystem feature such that files placed under public/fs are made available to the corresponding page via the BrowserFS `fs` module.

For example:

`public/fs/article/data.sqlite -> pages/article.js/data.sqlite`

Import statements could be used to determine if a file is prebaked into the page or lazily loaded.
Lazy loading on a per page basis could be controlled with a `fsload('./data.sqlite')` type function.

It gets interesting if you think about the cache and how that might be optimized. For example, next could convert all the files to their sha256sum name, store them in the browser cache and only perform an async lookup to the server if the file is missing. This would be great for content sites.

**Comment by james-pre** [comment]
Hi @ericbets, I'm the maintainer of BrowserFS. I'd like to let you know that BrowserFS has been deprecated. You can read more about that here: https://github.com/browser-fs/NOTICE. [ZenFS](https://github.com/zen-fs/core) is an updated and active fork that I created.
